### ISSUE TITLE
Clarify the status of shared_optimizer = 0

### ISSUE THREAD

**Issue by sleepinyourhat** [nyu-mll/jiant#597]
From the readme: "We have partial support for per-task optimizers (shared_optimizer = 0), but checkpointing may not behave correctly in this configuration. In the per-task case, we stop training on a task when its patience has run out or its optimizer hits the minimum learning rate."

If there's a reason people might want to use this, clean it up and explain it a bit more clearly. If not, let's just remove this option.

**Comment by sleepinyourhat** [comment]
Still on it? This is the last day I can do much reviewing before Mon.

**Comment by W4ngatang** [comment]
Resolved with #805
